src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/15.diff.sql

```diff
@@ -0,0 +1,88 @@
+﻿/*************************************************************
+    Event Agent checkpoint feature
+**************************************************************/
+
+/*************************************************************
+    Event Agent checkpoint table
+**************************************************************/
+
+IF NOT EXISTS (SELECT 1 FROM sys.tables WHERE name = 'EventAgentCheckpoint')
+BEGIN
+     CREATE TABLE dbo.EventAgentCheckpoint
+     (
+         CheckpointId varchar(64) NOT NULL,
```

Comment:
what's the strategy of assigning the checkpointId? This schema instance should be per database instance, and database instance should  be per FHIR account, if we currently only has one agent for the FHIR service at a time, how this column is used?

Reply:
I was thinking the CheckpointId could be a constant identifier set by the Event Agent in our first iteration. It is not totally needed if there will only be one Event agent. However the interfaces I had defined in healthcare-shared-components require an identifier currently because I was thinking of the future cases where there could be multiple Event Agents.

Reply:
Got it, thanks for explaining.

We would make sure that the checkpointId should be consistent if the EventAgent is down and restarted, otherwise, the checkpoint will not be picked up.